ngmix/joint_prior.py

```diff
@@ -17,184 +17,71 @@
-            p[w]=GMixND.get_prob_array(self,pars[w,:])
-
-        return p
-
-    def get_lnprob_array(self, pars, **keys):
-        """
-        log probability for array input [N,ndims]
-        """
-        lnp=zeros(pars.shape[0]) + LOWVAL
 
-        w=self.check_bounds_array(pars)
-        if w.size > 0:
-            lnp[w]=GMixND.get_lnprob_array(self,pars[w,:])
+    def __init__(self,
+                 cen_prior,
+                 g_prior,
+                 T_prior,
+                 F_prior):
 
-        return lnp
+        #print("JointPriorSimpleSep")
 
-    def sample(self, n=None):
-        """
-        Get samples for TF
-        """
+        self.cen_prior=cen_prior
+        self.g_prior=g_prior
+        self.T_prior=T_prior
 
-        if n is None:
-            is_scalar=True
-            n=1
+        if isinstance(F_prior,list):
+            self.nband=len(F_prior)
         else:
-            is_scalar=False
-
-        samples=zeros( (n,2) )
-
-        nleft=n
-        ngood=0
-        while nleft > 0:
-
-            tsamples=self._gmm.sample(nleft)
-            w=self.check_bounds_array(tsamples)
-
-            if w.size > 0:
-                first=ngood
-                last=ngood+w.size
-
-                samples[first:last,:] = tsamples[w,:]
-
-                ngood += w.size
-                nleft -= w.size
-
-        if is_scalar:
-            samples = samples[0,:]
-        return samples
-
-
-    def check_bounds_scalar(self, pars, throw=True):
-        """
-        Check bounds on T Flux
-        """
-
-        T=pars[0]
-        F=pars[1]
-
-        T_bounds=self.T_bounds
-        if T_bounds is not None:
-            if T < T_bounds[0] or T > T_bounds[1]:
-                if throw:
-                    raise GMixRangeError("T out of range")
-                else:
-                    return False
+            self.nband=1
+            F_prior=[F_prior]
 
-        F_bounds=self.F_bounds
-        if F_bounds is not None:
-            if F < F_bounds[0] or F > F_bounds[1]:
-                if throw:
-                    raise GMixRangeError("F out of range")
-                else:
-                    return False
+        self.F_priors=F_prior
 
-        return True
+        self.set_bounds()
 
-    def check_bounds_array(self,pars):
+    def set_bounds(self):
         """
-        Check bounds on T Flux
+        set possibe bounds
         """
-        T_bounds=self.T_bounds
-        F_bounds=self.F_bounds
-
-        T=pars[:,0]
-        F=pars[:,1]
-
-        logic = ones(T.shape[0], dtype=bool)
-        if T_bounds is not None:
-            T_logic = ( (T > T_bounds[0]) & (T < T_bounds[1]) )
-            logic = logic & T_logic
-
-        if F_bounds is not None:
-            F_logic = ((F > F_bounds[0]) & (F < F_bounds[1]) )
-            logic = logic & F_logic
+        bounds = [
+            (None,None), # c1
+            (None,None), # c2
+            (None,None), # g1
+            (None,None), # g2
```

Comment:
I guess the assumption here (and below) is that centers and shears never have bounds?

Reply:
yes, currently